controllers/datadogagent/feature/factory.go

```diff
@@ -57,6 +58,17 @@ func BuildFeatures(dda *v2alpha1.DatadogAgent, options *Options) ([]Feature, Req
 		requiredComponents.Merge(&reqComponents)
 	}
 
+	if dda.Spec.Global != nil &&
+		dda.Spec.Global.ContainerProcessStrategy != nil &&
+		dda.Spec.Global.ContainerProcessStrategy.Type == common.NonPrivilegedMultiProcessContainer &&
+		// This condition requires all features, which need agent to include it in required components.
+		// Otherwise test fails since this condition turn true when feature is disabled.
+		requiredComponents.Agent.IsEnabled() &&
+		!requiredComponents.Agent.IsPrivileged() {
```

Comment:
```suggestion
	if (
	    dda.Spec.Global != nil &&
	    dda.Spec.Global.ContainerProcessStrategy != nil &&
	    dda.Spec.Global.ContainerProcessStrategy.Type == common.NonPrivilegedMultiProcessContainer &&
	    // This condition requires that all features that need agent include it in the returned RequiredComponents.
	    // Otherwise, the tests fail since this condition becomes `true` when a feature is disabled.
	    requiredComponents.Agent.IsEnabled() &&
	    !requiredComponents.Agent.IsPrivileged() 
	    ) {
```

Formatting suggestion for readability. I also edited the comment line for grammar, but I don't understand what it means; do you mind clarifying?

Reply:
Linter doesn't like this formatting so had to undo it.

Reply:
ok here is a suggestion to clarify the comment:
```
// All features that need the NodeAgent must include it in their RequiredComponents; otherwise tests will fail when checking `requiredComponents.Agent.IsPrivileged()`.
```

Reply:
Thanks, added. this certainly sounds better!